Bottom: c0cf73cdfaee9d00ee452bd869307fdbb9325ae4
Top:    ff3ecff8378d9a045c1a91d0ad0d36e875de8d88
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-12-13 12:00:08 +0800

Refresh of virtio-net-correctly-enable

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index b425fa1..fe9f772 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1930,7 +1930,9 @@ static int virtnet_probe(struct virtio_device *vdev)
 		goto free_unregister_netdev;
 	}
 
-	virtnet_set_affinity(vi);
+	rtnl_lock();
+	virtnet_set_queues(vi, vi->curr_queue_pairs);
+	rtnl_unlock();
 
 	/* Assume link up if device can't report link status,
 	   otherwise get link status from config. */
